### PR TITLE
Part 2

### PR DESCRIPTION
**Pull request by hklerer**

### CODE REVIEW

File: src/main/java/klerer/rijks/RijksSearchFrame.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+public class RijksSearchFrame extends JFrame {
+    private RijksService rijksService;
+    private JTextField searchBar;
+    private JButton prevButton;
+    private JButton nextButton;
+    private JPanel imagePanel;
+    private int currentPage = 1;
+    ApiKey apiKey = new ApiKey();
+    String keyString = apiKey.get();
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksSearchFrame.java]: Private

**Comment by andrewoid** [comment]
You're missing a lot from part 2.

**Comment by hklerer** [comment]
i dont know how to do the handle response method

**Comment by andrewoid** [comment]
> i dont know how to do the handle response method

Well you have to try something...

File: src/main/java/klerer/rijks/ImageFrame.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class ImageFrame extends JFrame {
+    public ImageFrame(String title, String artist, String imageUrl) throws Exception {
+        setTitle(title + "by" + artist);
+        setSize(1050, 850);
+        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        URL url = new URL(imageUrl);
+        Image image = ImageIO.read(url);
+        Image scaledImage = image.getScaledInstance(700, -1, Image.SCALE_DEFAULT);
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/ImageFrame.java:17]: If the width of the image is 700, why are you making the size of the window 1050?

**Review by andrewoid — COMMENTED** [review]
Add a readme and screenshots

File: src/main/java/klerer/rijks/ImageFrame.java (hunk 2 above)
Comment (hklerer) [reply src/main/java/klerer/rijks/ImageFrame.java:17]: I changed it but i made it that way so you can see the image

File: src/main/java/klerer/rijks/RijksSearchFrame.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                    .subscribe(
+                            (response) -> handleResponse(response),
+                            Throwable::printStackTrace);
+        }
+    }
+
+    private void handleResponse(ArtObjects artObjects) {
+        imagePanel.removeAll();
+
+        if (artObjects.getArtObjects() == null) {
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksSearchFrame.java]: It might not be null, it might be an array of length 0

**Review by andrewoid — APPROVED** [review]

**Comment by hklerer** [comment]
Some things in my code is failing. When I search an artist, nothing happens and I get errors or when I search something that there is no results for the same occurs.

**Comment by hklerer** [comment]
Also it says you approved my PR but then you commented something else so is it approved or not?

**Comment by andrewoid** [comment]
> Some things in my code is failing. When I search an artist, nothing happens and I get errors or when I search something that there is no results for the same occurs.

Make sure the artist name is exact.

**Comment by hklerer** [comment]
I tried this name exactly and it did not work - Johannes Vermeer

File: src/main/java/klerer/rijks/RijksSearchFrame.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                    // tells Rx to request the data on a background Thread
+                    .subscribeOn(Schedulers.io())
+                    // tells Rx to handle the response on Swing's main Thread
+                    .observeOn(SwingSchedulers.edt())
+                    //.observeOn(AndroidSchedulers.mainThread()) // Instead use this on Android only
+                    .subscribe(
+                            (response) -> handleResponse(response),
+                            Throwable::printStackTrace);
+        } else {
+            disposable = rijksService.queryAndPageNum(keyString, searchBar.getText(), currentPage)
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksSearchFrame.java]: Use the other method on rijksService to search for the involvedMaker

Comment (hklerer) [reply src/main/java/klerer/rijks/RijksSearchFrame.java]: I should add an else if or I should change line 93?

Comment (andrewoid) [reply src/main/java/klerer/rijks/RijksSearchFrame.java]: Change the line. The search might work

Comment (hklerer) [reply src/main/java/klerer/rijks/RijksSearchFrame.java]: I tried changing it but its not working. It works to search an artist but not anything else and after the first search its very slow

Comment (andrewoid) [reply src/main/java/klerer/rijks/RijksSearchFrame.java]: > I tried changing it but its not working. It works to search an artist but not anything else and after the first search its very slow

I thought you said artist search wasn't working

Comment (hklerer) [reply src/main/java/klerer/rijks/RijksSearchFrame.java]: it wasn't but now artist search is working and its not working to search anything else

**Review by andrewoid — APPROVED** [review]

**Comment by hklerer** [comment]
I see that you approved my PR but I still have this error. Should I fix this?

**Comment by andrewoid** [comment]
> I see that you approved my PR but I still have this error. Should I fix this?

I send you an email.

**Pull request merged**